### PR TITLE
[ENH] Unit test for back converter from Gluonts to Multindex DF

### PR DESCRIPTION
**Pull request by TNTran92**
#### Reference Issues/PRs

Final portion of the solution to Issue #2860

#### What does this implement/fix? Explain your changes.

Implement unit test for the back convert from Gluonts's result to multiindex DF
Use `to_timestamp()` to go from period index to timestamp index`

#### Does your contribution introduce a new dependency? If yes, which one?

Dependency: gluon-ts

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [x] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [x] I've added unit tests and made sure they pass locally.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [x] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by TNTran92** [comment]
@fkiraly, I have checked and don't think it's possible to remove mxnet from gluonts. The reason is that MXNet to Gluon is like Keras to Tensorflow. 

Another reason is that, if we want to take advantage of the models of Gluonts,  we'll need Mxnet as soft dependency eventually since they are written in MXNet.

**Comment by fkiraly** [comment]
> Another reason is that, if we want to take advantage of the models of Gluonts, we'll need Mxnet as soft dependency eventually since they are written in MXNet.

I suppose this is true for using `gluonts`, but `MXNet` is a modelling library, and the data container should not require it.

Could you spell out the exact dependency requirements, and explain whether/why `MXNet` is required for the data container?

**Comment by TNTran92** [comment]
> > Another reason is that, if we want to take advantage of the models of Gluonts, we'll need Mxnet as soft dependency eventually since they are written in MXNet.
> 
> I suppose this is true for using `gluonts`, but `MXNet` is a modelling library, and the data container should not require it.
> 
> Could you spell out the exact dependency requirements, and explain whether/why `MXNet` is required for the data container?

There is no requirement as in `MXNet` has to be used for the data container. Even in `Gluonts`, `MXNet` is a soft dependency. I chose MXNet because most models in `Gluonts` are `MXNet`-based, so I thought it'd be convenient to use `MXNet` to produce an example of a typical estimator's output. There are other options, for example: `PyTorch`-based models and `Prophet` that can be used in the same manner. `Prophet` is already part of sktime, so maybe that is okay too.

Eventually, what I am trying to do is to get a representative sample from any Gluon estimator. Do you know of a better way this can be done?

**Comment by fkiraly** [comment]
>Eventually, what I am trying to do is to get a representative sample from any Gluon estimator. Do you know of a better way this can be done?

I do not know `gluonts` well enough, but surely (hopefully?) there is a way to construct the container without the need to import any soft dependency of `gluonts`, or any modelling package at all.

What I am concerned about is the coupling via imports. Generally we want to minimize unnecessary coupling, and coupling a data container to some modelling package that doesn't even use said data container seems very wrong to me.

**Comment by fkiraly** [comment]
FYI @TNTran92, @benHeid has found out how to handle soft dependencies in the `datatypes` module, see here: https://github.com/alan-turing-institute/sktime/pull/3255

**Comment by TNTran92** [comment]
Thanks, I will learn (steal) his method

**Comment by fkiraly** [comment]
>Thanks, I will learn (steal) his method

It's open source (permissive license) and this is a publicly visible record of credit, so it's not stealing :-)
